Code/Yahoo.Yui.Compressor.Tests/JavaScriptCompressorTaskTests.cs

```diff
@@ -182,6 +182,23 @@ public void An_Error_Is_Not_Logged_If_The_Output_File_Has_The_Same_Name_But_A_Di
             Assert.That(BuildEngine.DoesNotContainError(compressor.BuildEngine, ("Output file cannot be the same as source file(s).")));
         }
 
+        [Test]
+        public void If_OutputFile_Directory_Does_Not_Exist_It_Is_Created()
+        {
+            // Arange
+            var compressor = CreateCompressorTask();
+            compressor.SourceFiles = new ITaskItem[] { new TaskItem("Javascript Files\\SampleJavaScript1.js") };
+            compressor.OutputFile = "NewOutputDir\\Output.js";
+            if (Directory.Exists(Path.GetFullPath("NewOutputDir")))
+                Directory.Delete(Path.GetFullPath("NewOutputDir"), true);

```

Comment:
Please check that the directory was actually deleted after you attempt the delete.  If it failed silently then your test will pass whether or not there exists any code to actually create the folder.
Also please use {} for all if statements.